### PR TITLE
Cyber Implants +

### PR DESCRIPTION
**Pull request by xxalpha**
Made getting the more powerful implants significantly harder.

Modified finishing surgery step to heal brute damage if the patient has been sawed.

Added Nutriment Pump implants (regular and Plus). These will keep your burgermeter above Starvation or Hungry (if Plus). When EMPed they will inject poison instead of nutriment.

Added Reviver implant. This one will act as an automatic defibrillator when you die and also inject healing nanites. Takes time to recharge after implanting and after use. When EMPed it will need to recharge some and has a chance to cause a heart attack.

Edit 1: Medical HUDs can now detect cybernetic implants that have been implanted in humans.

Edit2: Added a Medical HUD implant to CMO locker.

---------

Chest implant sprite
implants

### CODE REVIEW

**Comment by WJohn** [comment]
Does this fix #5939?

**Comment by xxalpha** [comment]
I don't know? Likely not.

**Comment by GunHog** [comment]
While I understand making these harder to remove, I do not believe you should go so far as to make them so rare that they might as well not exist any more. 

From what I can tell, there are only three ways to get the biotech needed for and Rebooter Xray implants: The traitor item disguised as a health analyzer (which I have never seen), a xenomorph brain (good luck with that one!), and pyroclastic anomalies. It is not possible to work towards any of these items. For the first, you would have to get a traitor to give you his PDA code (or be a traitor yourself), kill and brain a xeno (which are very uncommon), and pyro anomalies.

May we not find a middle-ground here? They should be rare, but not on the same level as an average player's chance to become a wizard, or such.

**Comment by xxalpha** [comment]
The easiest way to get Biotech 6 that I figured out is deconstructing any syndicate item (toolbox because cheap), then printing a TYRANT board and deconstructing that, then deconstructing an Agent card (cheapest syndie = 3 tech I think). That will unlock the thermals. Then you deconstruct the thermals and you unlock all the OP shit.

**Comment by Lumipharon** [comment]
Gun, and research level that you can 'work towards' in reality means 'doable in 5 minutes.'
The ONLY difficulties in maxing research currently (ie: getting all levels high enough for every buildable item) is
A: waiting for the miners to get back
B: Getting a shotgun crate

The point of making OP shit only obtainable under uncommon circumstances is just that - to make them uncommon. Anything less basically means shit all, and they're way to powerful for that. Honestly stuff like the phazon should also be similarly difficult to get - it makes it a treat rather then shouting at miners to get lots of plasma.

**Comment by drynwyn** [comment]
Why is it that the nutriment pump implants have Biotech values less than their tech requirements? I get why from a balance standpoint, but it's inconsistent with the majority of R&D items- you should just lower their Biotech requirement.

File: code/modules/surgery/organs/cybernetic_implants.dm [hunk 1]
```diff
@@ -231,6 +242,121 @@
...
+		recharge_time = world.time + (2000 / severity)
+	else
+		recharge_time += 2000 / severity
+
+	if(prob(30/severity))
+		playsound(owner, 'sound/machines/defib_saftyOff.ogg', 50, 1, -1)
+		spawn(30)
+			playsound(owner, 'sound/machines/defib_zap.ogg', 50, 1, -1)
+			owner.heart_attack = 1
+			owner.visible_message("<span class = 'userdanger'>[owner] clutches at their chest as if their heart stopped!</span>")
```
Comment (Dorsisdwarf) [inline code/modules/surgery/organs/cybernetic_implants.dm]: Is this the same message as for the usual hear attacks? If not, it should be.
Also, it should check to see if someone's unconscious before saying that they clutch their chest.

Comment (xxalpha) [inline code/modules/surgery/organs/cybernetic_implants.dm]: Yes and you're right, I'll add a check.

Comment (GunHog) [inline code/modules/surgery/organs/cybernetic_implants.dm]: Heh, bloody pacemakers. :+1:

**Comment by Fox-McCloud** [comment]
DS should probably start out with all of these and have sunglasses instead of thermal goggles.

They're the freaking DS after all--you're not supposed to be able to beat them.

**Comment by Dorsisdwarf** [comment]
Well, you can totally beat the deathsquad, but you should have to damn well earn that victory with all of your robusting power, lobbing acid grenades, chucking minibombs, smashing them with mechs then hand-teleporting to safety when their buddies open fire, explosive shotgun darts, meteor shells - all these high-powered weaponry.

Clown and his PDA just shouldn't cut it.

**Comment by xxalpha** [comment]
@Jordie0608 got a sprite, this can now be merged, remove tag please.

**Comment by optimumtact** [comment]
no

**Pull request merged**